examples/hyperparamopt/hpo_example.py

```diff
@@ -0,0 +1,130 @@
+#  Copyright 2016 Intel Corporation
+#
+#  Licensed under the Apache License, Version 2.0 (the "License");
+#  you may not use this file except in compliance with the License.
+#  You may obtain a copy of the License at
+#
+#       http://www.apache.org/licenses/LICENSE-2.0
+#
+#  Unless required by applicable law or agreed to in writing, software
+#  distributed under the License is distributed on an "AS IS" BASIS,
+#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#  See the License for the specific language governing permissions and
+#  limitations under the License.
+
+"""Example for using hyperparameter optimization (hpo) package.
+
+In this example, we will try to optimize a function of
+2 variables (branin) using both hpo and grid search. 
+
+"""
+
+import brainiak.hyperparamopt.hpo as hpo
+import numpy as np
+import scipy.stats as st
+import matplotlib.pyplot as plt
+
+# Branin is the function we want to minimize.
+# It is a function of 2 variables
+def branin(x1, x2):
```

Comment:
@narayanan2004 Thanks, and sorry for not being clear: When I ask questions like "What does this mean? Is it someone's name? Why is this a good function for us?" I'm really pretending to be a user, these are the questions in my head. My experience of your code will be better if the answers are in the comments or docstrings. Actually answering my questions in Github only helps me, but putting the answers in the code would help many others.